### ISSUE TITLE
Error on making custom date filter

### ISSUE THREAD

**Issue by yelizariev** [odoo/odoo#64655]
Impacted versions: master, 14.0

Steps to reproduce:

* click ``Filters >> Add customer filters``
* choose any date field
* click on the date value
* remove that value
* click on the empty value

Current behavior:

```
Error:
Uncaught TypeError: date.format is not a function

https://5776787-master.runbot37.odoo.com/web/static/src/js/components/datepicker.js:124
Traceback:
TypeError: date.format is not a function
    at DateTimePicker._onDateTimePickerHide (https://5776787-master.runbot37.odoo.com/web/static/src/js/components/datepicker.js:124:39)
    at HTMLDivElement.dispatch (https://5776787-master.runbot37.odoo.com/web/static/lib/jquery/jquery.js:5183:27)
    at HTMLDivElement.elemData.handle (https://5776787-master.runbot37.odoo.com/web/static/lib/jquery/jquery.js:4991:28)
    at Object.trigger (https://5776787-master.runbot37.odoo.com/web/static/lib/jquery/jquery.js:8249:12)
    at HTMLDivElement.<anonymous> (https://5776787-master.runbot37.odoo.com/web/static/lib/jquery/jquery.js:8327:17)
    at Function.each (https://5776787-master.runbot37.odoo.com/web/static/lib/jquery/jquery.js:354:19)
    at jQuery.fn.init.each (https://5776787-master.runbot37.odoo.com/web/static/lib/jquery/jquery.js:189:17)
    at jQuery.fn.init.trigger (https://5776787-master.runbot37.odoo.com/web/static/lib/jquery/jquery.js:8326:15)
    at TempusDominusBootstrap4._notifyEvent (https://5776787-master.runbot37.odoo.com/web/static/lib/tempusdominus/tempusdominus.js:576:27)
    at TempusDominusBootstrap4.hide (https://5776787-master.runbot37.odoo.com/web/static/lib/tempusdominus/tempusdominus.js:2517:18)
```
Expected behavior:

no errors

Video/Screenshot link (optional):

https://drive.google.com/file/d/1gGvysbHVsZVr6iapBw6NRIZc4wLMXLGR/view?usp=sharing

Support ticket number submitted via odoo.com/help (optional):

**Comment by yelizariev** [comment]
@Arcasias @Polymorphe57

**Comment by yelizariev** [comment]
Comes from here

https://github.com/odoo/odoo/blob/9ffd6a62366ce75d125bcab8e1858a05b7c34c74/addons/web/static/src/js/components/datepicker.js#L122-L126

because may return ``false``

https://github.com/odoo/odoo/blob/9ffd6a62366ce75d125bcab8e1858a05b7c34c74/addons/web/static/src/js/components/datepicker.js#L105-L111

**Comment by yelizariev** [comment]
This fix
```
         _onDateTimePickerHide() {
             const date = this._parseInput(this.inputRef.el.value);
-            this.state.warning = date.format('YYYY-MM-DD') > moment().format('YYYY-MM-DD');
+            if (date) {
+                this.state.warning = date.format('YYYY-MM-DD') > moment().format('YYYY-MM-DD');
+            }
             this.trigger('datetime-changed', { date });
         }
```
gives another error:

```
Error: Invalid Prop 'date' in component 'DateTimePicker'
    at Object.QWeb.utils.validateProps (http://devel.14.localhost:14069/web/static/lib/owl/owl.js:3710:27)
    at DateTimePicker.__updateProps (http://devel.14.localhost:14069/web/static/lib/owl/owl.js:4258:32)
    at QWeb.eval (eval at _compile (http://devel.14.localhost:14069/web/static/lib/owl/owl.js:1695:28), <anonymous>:204:30)
    at QWeb.render (http://devel.14.localhost:14069/web/static/lib/owl/owl.js:1617:32)
    at Object.__owl__.renderFn (http://devel.14.localhost:14069/web/static/src/js/model.js:482:20)
    at PatchableClass.__render (http://devel.14.localhost:14069/web/static/lib/owl/owl.js:4354:37)
    at http://devel.14.localhost:14069/web/static/lib/owl/owl.js:4103:26
```

**Comment by sswapnesh** [comment]
Seems duplicate of https://github.com/odoo/odoo/issues/42435

**Comment by msh-odoo** [comment]
@yelizariev Should be fixed with #58092
